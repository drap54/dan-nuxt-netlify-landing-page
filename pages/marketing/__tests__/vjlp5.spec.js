import { mount } from '@vue/test-utils'

import index from '@/pages/marketing/vjlp5/_slug/index'

describe('vjlp5 page', () => {
  let wrapper 
  beforeEach(async () => {
    wrapper = mount(index)
    const data = await wrapper.vm.$options.asyncData({
      params: { slug: '105-vj-lan-jp-pro-mobile-homepage-a-1219' }
    })
    await wrapper.setData({ dataMD: data });
  })
  
  test('mounts properly', () => {
    expect(wrapper.vm).toBeTruthy()
  })

  test('renders properly', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('emitMDcontent event emits properly', async () => {
    wrapper.vm.$emit('emitMDcontent', { foo: 'bar' })
    await wrapper.vm.$nextTick()
    expect(wrapper.emitted().emitMDcontent[1]).toEqual([{ foo: 'bar' }])
  })

  test('dataMD async data should not be null', () => {
    expect(wrapper.vm.dataMD).not.toBeNull()
  })
})