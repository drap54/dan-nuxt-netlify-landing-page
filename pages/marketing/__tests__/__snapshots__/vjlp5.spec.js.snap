// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`vjlp5 page renders properly 1`] = `"<div></div>"`;
